classes/Cart.php

```diff
@@ -138,6 +141,7 @@ class CartCore extends ObjectModel
             'allow_seperated_package' => ['type' => self::TYPE_BOOL, 'validate' => 'isBool'],
             'date_add' => ['type' => self::TYPE_DATE, 'validate' => 'isDate'],
             'date_upd' => ['type' => self::TYPE_DATE, 'validate' => 'isDate'],
+            'reference_order' => ['type' => self::TYPE_STRING, 'allow_null' => true],
```

Comment:
:warning: self::TYPE_STRING but the PHPDoc is `int`.

Reply:
If it's a string, define the size

Reply:
After all the discussion with @kpodemski I'm still not 100% sure about using this reference

Split order shouldn't be a problem since they share the same reference anyway, and for this use case we actually don't care about the order reference since we don't use it It is only used to know if the cart is ordered or not

But once this field is introduce some other devs may start using it so if it is not consistent it might lead to some unexpected behaviour so we need to be careful about it

So I'm thinking maybe a simple boolean would be enough for us here, and less confusing Like just a `ordered` boolean to avoid calling `Order::getIdByCartId` for performance issue

Reply:
@jolelievre `ordered` flag is ok, also it's a handy way to get all non ordered carts from for example 2 years ago and just throw them into space :) one suggestion would be to have some helper methods like `static::isCartOrdered` `$cart->markAsOrdered()`

Reply:
I'm not sure but I believe such detection of abandoned carts already exist in `AdminStatsController::getAbandonedCarts` or in `Cart::lastNoneOrderedCart` (there might be other similar features)

It's already doable easily with a leftJoin on order table

Reply:
I won't touch the db structure (as discussed on slack, it's not worth it for now)